lib/internal/webidl.js

```diff
@@ -236,37 +246,98 @@ function createEnumConverter(name, values) {
 
 // https://tc39.es/ecma262/#sec-ecmascript-data-types-and-values
 function type(V) {
-  if (V === null)
-    return 'Null';
-
   switch (typeof V) {
     case 'undefined':
-      return 'Undefined';
+      return UNDEFINED;
     case 'boolean':
-      return 'Boolean';
+      return BOOLEAN;
     case 'number':
-      return 'Number';
+      return NUMBER;
     case 'string':
-      return 'String';
+      return STRING;
     case 'symbol':
-      return 'Symbol';
+      return SYMBOL;
     case 'bigint':
-      return 'BigInt';
+      return BIGINT;
     case 'object': // Fall through
     case 'function': // Fall through
     default:
+      if (V === null) {
+        return NULL;
+      }
       // Per ES spec, typeof returns an implementation-defined value that is not
       // any of the existing ones for uncallable non-standard exotic objects.
       // Yet Type() which the Web IDL spec depends on returns Object for such
       // cases. So treat the default case as an object.
-      return 'Object';
+      return OBJECT;
   }
 }
 
+// https://webidl.spec.whatwg.org/#js-dictionary
+function createDictionaryConverter(members) {
+  // The spec requires us to operate the members of a dictionary in
+  // lexicographical order. We are doing this in the outer scope to
+  // reduce the overhead that could happen in the returned function.
+  const sortedMembers = ArrayPrototypeToSorted(members, (a, b) => {
+    if (a.key === b.key) {
+      return 0;
+    }
+    return a.key < b.key ? -1 : 1;
+  });
+
+  return function(
+    V,
+    opts = kEmptyObject,
+  ) {
+    if (V != null && type(V) !== OBJECT) {
+      throw makeException(
+        'can not be converted to a dictionary',
+        opts,
+      );
+    }
+
+    const idlDict = { __proto__: null };
+    for (let i = 0; i < sortedMembers.length; i++) {
+      const member = sortedMembers[i];
+      const key = member.key;
+      let jsMemberValue;
+      if (V == null) {
+        jsMemberValue = undefined;
+      } else {
+        jsMemberValue = V[key];
+      }
+
+      if (jsMemberValue !== undefined) {
+        const memberContext = opts.context ? `${key} in ${opts.context}` : `${key}`;
+        const converter = member.converter;
+        const idlMemberValue = converter(
+          jsMemberValue,
+          {
+            __proto__: null,
+            prefix: opts.prefix,
+            context: memberContext,
+          },
+        );
+        idlDict[key] = idlMemberValue;
+      } else if (typeof member.defaultValue === 'function') {
+        const idlMemberValue = member.defaultValue();
+        idlDict[key] = idlMemberValue;
+      } else if (member.required) {
+        throw makeException(
+          `can not be converted because of the missing '${key}'`,
```

Comment:
```suggestion
          `cannot be converted because of the missing '${key}'`,
```

"Can not, as two separate words, is not a proper verb phrase. When describing an action which is impossible, you should always use cannot." https://writingexplained.org/cannot-or-can-not-difference

Reply:
Thanks. Done